tools.go

```diff
@@ -0,0 +1,7 @@
+// +build tools
+
+package tools
+
+import (
+	_ "golang.org/x/tools/cmd/goimports"
```

Comment:
I didn't quite understand why this file has to exist in perpetuity, since `go.mod` has already been modified to pick up the goimports dependency, but this seems to be the recommended best practice in [this thread](https://github.com/golang/go/issues/25922#issuecomment-402918061).

Reply:
I don't understand this either.  I'm trying to follow best practices but I'm having such a hard time finding any documentation for the go modules behavior that it's very hard to tell what this does if it's necessary.  Let me do some experimentation...